### PR TITLE
Allows users to set up a Do not Disturb message when in meetings in embedded mode

### PR DESCRIPTION
**Pull request by maisnamrajusingh**
#### Summary
Allows users to set up a Do not Disturb message when in meetings in embedded mode. This wouldn't work in case of meetings where the user clicks the link and opens the jisti window in another window. 

#### Ticket Link
https://github.com/mattermost/mattermost-plugin-jitsi/issues/143

### CODE REVIEW

**Review by larkox — COMMENTED** [review]
A few questions about the implementation.

**Comment by jasonblais** [comment]
cc @maisnamrajusingh about the feedback above. Thanks!

File: webapp/src/actions/index.ts [hunk 1]
```diff
@@ -94,3 +94,22 @@ export function openJitsiMeeting(post: Post | null, jwt: string | null): ActionF
         return {data: null};
     };
 }
+
+export function setUserStatus(userId: string | null, status: string): ActionFunc {
```
[the remark this answers is not in the document]
Comment (maisnamrajusingh) [reply webapp/src/actions/index.ts]: @larkox  I tried using the setStatus but wasn't able to get it working with that hence I stuck with this one

File: webapp/src/components/conference/conference.tsx [hunk 2]
```diff
@@ -163,6 +164,11 @@ export default class Conference extends React.PureComponent<Props, State> {
         this.api.executeCommand('hangup');
         setTimeout(() => {
             this.props.actions.openJitsiMeeting(null, null);
+            var userId = null;
+            if (this.props.post) {
+                userId = this.props.post.user_id;
+            }
+            this.props.actions.setUserStatus(userId, 'online');
```
[the remark this answers is not in the document]
Comment (maisnamrajusingh) [reply webapp/src/components/conference/conference.tsx]: yup, it doesn't make sense. I am getting the userId from redux now.

File: webapp/src/actions/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (larkox) [reply webapp/src/actions/index.ts]: What is the problem you are facing with the setStatus action?

[the remark this answers is not in the document]
Comment (maisnamrajusingh) [reply webapp/src/actions/index.ts]: another reason was that the plugin is already using a lot of code in the client, so  I wanted it to be consistent with the current codebase.

File: webapp/src/components/conference/conference.tsx [hunk 3]
```diff
@@ -163,6 +164,7 @@ export default class Conference extends React.PureComponent<Props, State> {
         this.api.executeCommand('hangup');
         setTimeout(() => {
             this.props.actions.openJitsiMeeting(null, null);
+            this.props.actions.setUserStatus(this.props.currentUserId, 'online');
```
Comment (hanzei) [inline webapp/src/components/conference/conference.tsx:167]: Is there a constant that can be used here?

**Review by hanzei — APPROVED** [review]
LGTM :+1: Tested and works as expected

**Review by larkox — APPROVED** [review]

File: webapp/src/components/conference/conference.tsx (hunk 3 above)
Comment (maisnamrajusingh) [reply webapp/src/components/conference/conference.tsx:167]: I actually wanted to use an enum for this. Does that make sense ?

Comment (hanzei) [reply webapp/src/components/conference/conference.tsx:167]: @maisnamrajusingh Can you use the one define in mattermost-redux: https://github.com/mattermost/mattermost-redux/blob/40368bb807b8b5a3dc0f4ce4a5a76eabc5c9de54/src/constants/general.ts#L21-L21

**Comment by hanzei** [comment]
Blocking until https://github.com/mattermost/mattermost-plugin-jitsi/pull/195#discussion_r662191398 is resolved.

File: webapp/src/components/conference/conference.tsx (hunk 3 above)
Comment (jasonblais) [reply webapp/src/components/conference/conference.tsx:167]: @maisnamrajusingh please see above

**Review by hanzei — APPROVED** [review]

**Pull request merged**